packages/roosterjs-editor-api/lib/format/getFormatState.ts

```diff
@@ -20,13 +20,24 @@ export function getElementBasedFormatState(
     event?: PluginEvent
 ): ElementBasedFormatState {
     let listTag = getTagOfNode(editor.getElementAtCursor('OL,UL', null /*startFrom*/, event));
+    let range = editor.getSelectionRange();
+
+    // Check if selection is multiline, spans more than one block
+    let multiline = false;
+    if (range) {
+        let startingBlock = editor.getBlockElementAtNode(range.startContainer);
+        let endingBlock = editor.getBlockElementAtNode(range.endContainer);
+        multiline = endingBlock.equals(startingBlock);
+    }

```

Comment:
As an optimization, you can first check if range is collapsed (range.collapsed), when it is true, we know it must not be multiple line, and no need to create block element. This is a the most common case and it is fast enough.